### PR TITLE
Replace gcp hardcoded values with generic annotations

### PR DESCRIPTION
**Pull request by ChethanUK**
## Description
Instead of hardcoded GCP load balancer values replacing them with generic annotation map.

### CODE REVIEW

File: kubernetes/helm/pinot/templates/broker/service-external.yaml [hunk 1]
```diff
@@ -24,10 +24,7 @@ kind: Service
 metadata:
   name: {{ template "pinot.broker.external" . }}
   annotations:
-    type: "service-external"
-  {{- if .Values.broker.service.gcpInternalLB }}
-    cloud.google.com/load-balancer-type: Internal
-  {{- end }}
+{{ toYaml .Values.controller.serviceAnnotationsExternal | indent 4 }}
```
Comment (daniellavoie) [inline kubernetes/helm/pinot/templates/broker/service-external.yaml:27]: `.Values.broker.external.annotations`

File: kubernetes/helm/pinot/templates/controller/service.yaml [hunk 2]
```diff
@@ -22,10 +22,7 @@ kind: Service
 metadata:
   name: {{ include "pinot.controller.fullname" . }}
   annotations:
-    type: "service"
-  {{- if .Values.controller.service.gcpInternalLB }}
-    cloud.google.com/load-balancer-type: Internal
-  {{- end }}
+{{ toYaml .Values.controller.serviceAnnotations | indent 4 }}
```
Comment (daniellavoie) [inline kubernetes/helm/pinot/templates/controller/service.yaml:25]: `.Values.controller.service.annotations`

File: kubernetes/helm/pinot/values.yaml [hunk 3]
```diff
@@ -117,6 +115,14 @@ controller:
 
   podAnnotations: {}
 
+  # For example, in private GKE cluster, you might add cloud.google.com/load-balancer-type: Internal
+  serviceAnnotations:
+    type: "service"
```
Comment (daniellavoie) [inline kubernetes/helm/pinot/values.yaml:120]: `controller.service.annotations` already service that purpose. Also `type: "service"` and should not be present by default.

File: kubernetes/helm/pinot/values.yaml [hunk 4]
```diff
@@ -194,6 +198,14 @@ broker:
...
+
+  # serviceAnnotationsExternal: {}
+  serviceAnnotationsExternal:
+    type: "service-external"
+
```
Comment (daniellavoie) [inline kubernetes/helm/pinot/values.yaml:208]: Same observations as the controller configs.

File: kubernetes/helm/pinot/values.yaml [hunk 5]
```diff
@@ -353,6 +367,10 @@ minion:
 
   podAnnotations: {}
 
+  # For example, in private GKE cluster, you might add cloud.google.com/load-balancer-type: Internal
+  serviceAnnotations:
+    type: "service"
+
```
Comment (daniellavoie) [inline kubernetes/helm/pinot/values.yaml:373]: Same observations as the controller configs.

**Review by daniellavoie — CHANGES_REQUESTED** [review]
Thank you very much for the contribution @ChethanUK ! Couple of observations and little tweaks are required. `external` and `annotations` already exists in `values.yaml` for all service. it would be preferable to reuse them. 

More importantly, all configuration refers to controller values for all components. That's a problem, they should all refer to their respective components.

Finally, `annotations.type: "service"` does nothing on Kubernetes so I wouldn't add that as a default value.

File: kubernetes/helm/pinot/templates/broker/service-external.yaml (hunk 1 above)
Comment (ChethanUK) [reply kubernetes/helm/pinot/templates/broker/service-external.yaml:27]: Copy-paste errors. Fixing.

**Review by daniellavoie — APPROVED** [review]
LGTM!

@xiangfu0 mind reviewing too?

**Review by xiangfu0 — APPROVED** [review]

**Pull request merged**